engines/bastion_katello/app/assets/javascripts/bastion_katello/common/check-organization.service.js

```diff
@@ -0,0 +1,21 @@
+(function () {
+    'use strict';
+
+    /**
+     * @ngdoc service
+     * @name  Bastion.common.service:CheckOrganizationHandler
+     *
+     * @description
+     *   Provides common functionality in handling Resource/Organization warnings
+     */
+    function CheckOrganizationHandler(CurrentOrganization, Notification, translate) {
```

Comment:
Now that we have messaging coming out of the API controller I am wondering if we should include this any longer. It kind of introduces technical debt since we ideally handle this at the API now.

I also found that we have something in bastion_katello called 'ApiErrorHandler' and we can make it work with your API change like this:

```
        function handleError(response, $scope, defaultErrorMessage) {                                                                                                                         
            var errors = (response.data && response.data.errors) ? response.data.errors : [];                                                                                                 
            var error = (response.data && response.data.error) ? response.data.error : null;                                                                                                  
                                                                                                                                                                                              
            if (errors.length > 0) {                                                                                                                                                          
                angular.forEach(errors, function (error) {                                                                                                                                    
                    Notification.setErrorMessage(error);                                                                                                                                      
                });                                                                                                                                                                           
            } else if (error && response.status >= 400 && response.status <= 404) {                                                                                                           
                Notification.setErrorMessage(error.message);                                                                                                                                  
            } else {                                                                                                                                                                          
                Notification.setErrorMessage(defaultErrorMessage);                                                                                                                            
            }                                                                                                                                                                                 
                                                                                                                                                                                              
            if ($scope && $scope.hasOwnProperty('panel')) {                                                                                                                                   
                $scope.panel.error = true;                                                                                                                                                    
            }                                                                                                                                                                                 
        } 
```

This way, for 400 and 404 errors we can show the real message. I think that's a lot more informative than 'Something went wrong while retrieving the resource' but we still fall back to it.

I realize this will not achieve the same effect as your CheckOrganizationService because we don't pass the organization_id parameter everywhere in the UI. I think it would be worthwhile to:

- remove the CheckOrganizationService
- improve the ApiErrorHandler per my above example
- wherever you've added CheckOrganizationService, check if we are passing organization_id. If not, let's file an issue to fix those areas in 3.16. I'd be open to fixing them here, but up to you.

Reply:
I was testing this and it doesn't seem like this refactoring is needed. We do pass the api error message to the handleError method and it displays accordingly.